--- pgadmin3-1.14.0/acinclude.m4~	2011-09-17 10:05:51.000000000 +0200
+++ pgadmin3-1.14.0/acinclude.m4	2011-09-17 10:09:19.127611445 +0200
@@ -69,41 +69,9 @@
 ####################
 AC_DEFUN([LOCATE_WXWIDGETS],
 [
-	AC_ARG_WITH(wx, [  --with-wx=DIR		root directory for wxWidgets installation],
-	[
-		if test "$withval" != no
-		then
-			WX_HOME="$withval"
-			if test ! -f "${WX_HOME}/bin/wx-config"
-			then
-				AC_MSG_ERROR([Could not find your wxWidgets installation in ${WX_HOME}])
-			fi
-
-		fi
-		WX_CONFIG=${WX_HOME}/bin/wx-config
-	], 
-	[
-		WX_HOME=/usr/local/wx2
-		if test ! -f "${WX_HOME}/bin/wx-config"
-		then
-			WX_HOME=/usr/local
-			if test ! -f "${WX_HOME}/bin/wx-config"
-			then
-				WX_HOME=/usr
-				if test ! -f "${WX_HOME}/bin/wx-config"
-				then
-		    # Search the path
-				    AC_PATH_PROGS(WX_CONFIG, wx-config)
-		    if test ! -f "${WX_CONFIG}"
-				    then
-			AC_MSG_ERROR([Could not find your wxWidgets installation. You might need to use the --with-wx=DIR configure option])
-		    else
-					   WX_HOME=`${WX_CONFIG} --prefix`
-		    fi
-				fi
-			fi
-		fi
-		WX_CONFIG=${WX_HOME}/bin/wx-config
+	WX_HOME=/usr
+	AC_ARG_WITH(wx-config, [  --with-wx-config=FILE wx-config file],[
+	WX_CONFIG="$withval"
 	])
 ])
 
--- pgadmin3-1.14.0/acinclude.m4~	2011-09-17 10:09:39.000000000 +0200
+++ pgadmin3-1.14.0/acinclude.m4	2011-09-17 10:13:44.928871229 +0200
@@ -475,6 +475,7 @@
 		else
 			PG_SSL="no"
 		fi
+		PG_SSL=yes
 
 		if test "$LIB_KRB5" = "yes"
 		then
@@ -512,6 +513,7 @@
 		else
 			PG_KRB5="no"
 		fi
+		PG_KRB5=yes
 
 		AC_LANG_RESTORE
 
