diff -Nurb o.pgadmin3-1.2.2/src/Makefile.am n.pgadmin3-1.2.2/src/Makefile.am
--- o.pgadmin3-1.2.2/src/Makefile.am	2005-05-26 14:54:51.000000000 +0200
+++ n.pgadmin3-1.2.2/src/Makefile.am	2005-09-03 02:50:10.533940200 +0200
@@ -180,9 +180,9 @@
 if !APPBUNDLE
 
 nobase_dist_pkgdata_DATA = $(TMP_ui)
-AM_CXXFLAGS = -DDATA_DIR=\"$(pkgdatadir)/\" -Wall -g -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
-AM_CPPFLAGS = -DDATA_DIR=\"$(pkgdatadir)/\" -Wall -g -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
-AM_CFLAGS = -DDATA_DIR=\"$(pkgdatadir)/\" -Wall -g -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
+AM_CXXFLAGS = -DDATA_DIR=\"$(pkgdatadir)/\" -Wall -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
+AM_CPPFLAGS = -DDATA_DIR=\"$(pkgdatadir)/\" -Wall -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
+AM_CFLAGS = -DDATA_DIR=\"$(pkgdatadir)/\" -Wall -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
 
 # Automake trys to execute install-exec-hook if it appears anywhere in the file, so we need a dummy
 # for non-APPBUNDLE cases.
@@ -191,9 +191,9 @@
 else
 
 nobase_dist_data_DATA = $(TMP_ui)
-AM_CXXFLAGS = -Wall -g -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
-AM_CPPFLAGS = -Wall -g -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
-AM_CFLAGS = -Wall -g -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
+AM_CXXFLAGS = -Wall -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
+AM_CPPFLAGS = -Wall -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
+AM_CFLAGS = -Wall -I$(top_srcdir)/src/include -I$(top_srcdir)/src/agent/include
 
 install-exec-hook:
 	cd $(bindir) ;\
