diff -urN pgadmin3-1.6.1.org/acinclude.m4 pgadmin3-1.6.1/acinclude.m4
--- pgadmin3-1.6.1.org/acinclude.m4	2006-11-30 11:48:45.000000000 +0100
+++ pgadmin3-1.6.1/acinclude.m4	2006-12-08 01:22:54.341264000 +0100
@@ -47,34 +47,10 @@
 ####################
 AC_DEFUN([LOCATE_WXWIDGETS],
 [
-	AC_ARG_WITH(wx, [  --with-wx=DIR	   root directory for wxWidgets installation],
+	WX_HOME=/usr
+	AC_ARG_WITH(wx-config, [  --with-wx-config=FILE	wx-config file],
 	[
-		if test "$withval" != no
-		then
-			WX_HOME="$withval"
-						if test ! -f "${WX_HOME}/bin/wx-config"
-						then
-								AC_MSG_ERROR([Could not find your wxWidgets installation in ${WX_HOME}])
-						fi
-
-		fi
-		WX_CONFIG=${WX_HOME}/bin/wx-config
-	], 
-	[
-		WX_HOME=/usr/local/wx2
-		if test ! -f "${WX_HOME}/bin/wx-config"
-		then
-			WX_HOME=/usr/local
-			if test ! -f "${WX_HOME}/bin/wx-config"
-			then
-				WX_HOME=/usr
-				if test ! -f "${WX_HOME}/bin/wx-config"
-				then
-					AC_MSG_ERROR([Could not find your wxWidgets installation. You might need to use the --with-wx=DIR configure option])
-				fi
-			fi
-		fi
-		WX_CONFIG=${WX_HOME}/bin/wx-config
+		WX_CONFIG="$withval"
 	])
 ])
 
@@ -320,33 +296,8 @@
 		AC_LANG_C
 		AC_CHECK_LIB(pq, PQexec, [PG_LIBPQ=yes], [PG_LIBPQ=no])
 
-		if test "$build_cpu-$build_vendor" = "powerpc-apple"
-		then
-			echo -n "checking if libpq links against libssl: "
-			if test "$(otool -L ${PG_HOME}/lib/libpq.?.dylib | grep -c libssl)" -gt 0
-			then
-				PG_SSL="yes"
-			else
-				PG_SSL="no"
-			fi
-			echo $PG_SSL
-		else
-			AC_CHECK_LIB(pq, SSL_connect, [PG_SSL=yes], [PG_SSL=no])
-		fi
-
-		if test "$build_cpu-$build_vendor" = "powerpc-apple"
-		then
-			echo -n "checking if libpq links against libkrb5: "
-			if test "$(otool -L ${PG_HOME}/lib/libpq.?.dylib | grep -c libkrb5)" -gt 0
-			then
-				PG_KRB5="yes"
-			else
-				PG_KRB5="no"
-			fi
-			echo $PG_KRB5
-		else
-			AC_CHECK_LIB(pq, krb5_free_principal, [PG_KRB5=yes], [PG_KRB5=no])
-		fi
+		PG_SSL=yes
+		PG_KRB5=yes
 
 		AC_LANG_RESTORE
 
