--- pgadmin3/acinclude.m4.org	2007-08-28 12:50:06.399047988 +0200
+++ pgadmin3/acinclude.m4	2007-08-28 12:50:52.597961689 +0200
@@ -440,63 +440,8 @@
 		AC_LANG_SAVE
 		AC_LANG_C
 
-                # Check for SSL support
-                if test "$BUILD_STATIC" = "yes"
-                then
-                        AC_MSG_CHECKING(for SSL_connect in libpq.a)
-                        if test "$(nm ${PG_LIB}/libpq.a | grep -c SSL_connect)" -gt 0
-                        then
-                                AC_MSG_RESULT(present)
-                                PG_SSL="yes"
-                        else   
-                                AC_MSG_RESULT(not present)
-                                PG_SSL="no"
-                        fi
-                else
-		        if test "$build_cpu-$build_vendor" = "powerpc-apple" -o "$build_cpu-$build_vendor" = "i686-apple"
-        		then
-                                AC_MSG_CHECKING(for SSL_connect in -lpq)
-		        	if test "$(otool -L ${PG_LIB}/libpq.?.dylib | grep -c libssl)" -gt 0
-			        then
-                                        AC_MSG_RESULT(present)
-        				PG_SSL="yes"
-	        		else
-                                        AC_MSG_RESULT(not present)
-	        			PG_SSL="no"
-		        	fi
-        		else
-	        		AC_CHECK_LIB(pq, SSL_connect, [PG_SSL=yes], [PG_SSL=no])
-	        	fi
-                fi
-
-                # Check for Kerberos support
-                if test "$BUILD_STATIC" = "yes"
-                then
-                        AC_MSG_CHECKING(for krb5_free_principal in libpq.a)
-                        if test "$(nm ${PG_LIB}/libpq.a | grep -c krb5_free_principal)" -gt 0
-                        then
-                                AC_MSG_RESULT(present)
-                                PG_KRB5="yes"
-                        else
-                                AC_MSG_RESULT(not present)
-                                PG_KRB5="no"
-                        fi
-                else
-		        if test "$build_cpu-$build_vendor" = "powerpc-apple" -o "$build_cpu-$build_vendor" = "i686-apple"
-		        then
-                                AC_MSG_CHECKING(for krb5_free_principle in -lpq)
-			        if test "$(otool -L ${PG_LIB}/libpq.?.dylib | grep -c libkrb5)" -gt 0
-			        then
-                                        AC_MSG_RESULT(present)
-				        PG_KRB5="yes"
-			        else
-                                        AC_MSG_RESULT(not present)
-				        PG_KRB5="no"
-			        fi
-		        else
-			        AC_CHECK_LIB(pq, krb5_free_principal, [PG_KRB5=yes], [PG_KRB5=no])
-		        fi
-                fi
+		PG_SSL="yes"
+		PG_KRB5="yes"
 
 		AC_LANG_RESTORE
 
--- pgadmin3/acinclude.m4.org	2007-08-28 12:53:35.228151478 +0200
+++ pgadmin3/acinclude.m4	2007-08-28 12:54:31.519665591 +0200
@@ -108,41 +108,9 @@
 ####################
 AC_DEFUN([LOCATE_WXWIDGETS],
 [
-	AC_ARG_WITH(wx, [  --with-wx=DIR	   root directory for wxWidgets installation],
-	[
-		if test "$withval" != no
-		then
-			WX_HOME="$withval"
-						if test ! -f "${WX_HOME}/bin/wx-config"
-						then
-								AC_MSG_ERROR([Could not find your wxWidgets installation in ${WX_HOME}])
-						fi
-
-		fi
-		WX_CONFIG=${WX_HOME}/bin/wx-config
-	], 
-	[
-		WX_HOME=/usr/local/wx2
-		if test ! -f "${WX_HOME}/bin/wx-config"
-		then
-			WX_HOME=/usr/local
-			if test ! -f "${WX_HOME}/bin/wx-config"
-			then
-				WX_HOME=/usr
-				if test ! -f "${WX_HOME}/bin/wx-config"
-				then
-                    # Search the path
-				    AC_PATH_PROGS(WX_CONFIG, wx-config)
-                    if test ! -f "${WX_CONFIG}"
-				    then
-                        AC_MSG_ERROR([Could not find your wxWidgets installation. You might need to use the --with-wx=DIR configure option])
-                    else
-					   WX_HOME=`${WX_CONFIG} --prefix`
-                    fi
-				fi
-			fi
-		fi
-		WX_CONFIG=${WX_HOME}/bin/wx-config
+	WX_HOME=/usr
+	AC_ARG_WITH(wx-config, [  --with-wx-config=FILE wx-config file],[
+	WX_CONFIG="$withval"
 	])
 ])
 
