diff -ur pgadmin3-1.6.1.org/src/frm/frmEditGrid.cpp pgadmin3-1.6.1/src/frm/frmEditGrid.cpp
--- pgadmin3-1.6.1.org/src/frm/frmEditGrid.cpp	2006-11-30 11:48:30.000000000 +0100
+++ pgadmin3-1.6.1/src/frm/frmEditGrid.cpp	2006-12-08 01:06:14.686789500 +0100
@@ -75,7 +75,7 @@
     EVT_GRID_EDITOR_SHOWN(      frmEditGrid::OnEditorShown)
     EVT_GRID_EDITOR_HIDDEN(     frmEditGrid::OnEditorHidden)
     EVT_GRID_LABEL_RIGHT_CLICK( frmEditGrid::OnLabelRightClick)
-    EVT_AUI_PANEBUTTON(         frmEditGrid::OnAuiUpdate)
+    EVT_AUI_PANE_BUTTON(         frmEditGrid::OnAuiUpdate)
 END_EVENT_TABLE()
 
 
diff -ur pgadmin3-1.6.1.org/src/frm/frmQuery.cpp pgadmin3-1.6.1/src/frm/frmQuery.cpp
--- pgadmin3-1.6.1.org/src/frm/frmQuery.cpp	2006-11-30 11:48:30.000000000 +0100
+++ pgadmin3-1.6.1/src/frm/frmQuery.cpp	2006-12-08 01:13:41.662723750 +0100
@@ -103,7 +103,7 @@
     EVT_ACTIVATE(                   frmQuery::OnActivate)
     EVT_STC_MODIFIED(CTL_SQLQUERY,  frmQuery::OnChangeStc)
     EVT_STC_UPDATEUI(CTL_SQLQUERY,  frmQuery::OnPositionStc)
-    EVT_AUI_PANECLOSE(              frmQuery::OnAuiUpdate)
+    EVT_AUI_PANE_CLOSE(              frmQuery::OnAuiUpdate)
 END_EVENT_TABLE()
 
 frmQuery::frmQuery(frmMain *form, const wxString& _title, pgConn *_conn, const wxString& query)
diff -ur pgadmin3-1.6.1.org/src/main/events.cpp pgadmin3-1.6.1/src/main/events.cpp
--- pgadmin3-1.6.1.org/src/main/events.cpp	2006-11-30 11:48:30.000000000 +0100
+++ pgadmin3-1.6.1/src/main/events.cpp	2006-12-08 01:13:52.807420250 +0100
@@ -71,7 +71,7 @@
     EVT_STC_UPDATEUI(CTL_SQLPANE,           frmMain::OnPositionStc)
     EVT_CLOSE(                              frmMain::OnClose)
 
-    EVT_AUI_PANECLOSE(                      frmMain::OnAuiUpdate)
+    EVT_AUI_PANE_CLOSE(                      frmMain::OnAuiUpdate)
 
 #ifdef __WXGTK__
     EVT_TREE_KEY_DOWN(CTL_BROWSER,          frmMain::OnTreeKeyDown)
